### DISCUSSION TITLE
Timing or order of pressed / released / held

### DISCUSSION THREAD

**Discussion by Torxsmind** [gpiozero/gpiozero#1081] [Q&A]
Hello, I am trying to have a simple script for my raspberry PI to either shutdown when a GPIO button is held for more than x Seconds.  OR reboot when a momentary press.

Here is the script I have so far:
`
from gpiozero import Button
from signal import pause
import os

def reboot():
    os.system('sudo reboot')

def shutdown():
    os.system('sudo shutdown -h now')

button = Button(16)

button.when_pressed = reboot
button.when_held = shutdown

pause()
`

No matter how long I press / hold it always reboots the PI.  No shutdown.

Thoughts?

**Comment by bennuttall** [comment]
This is covered in the FAQ: https://gpiozero.readthedocs.io/en/stable/faq.html#how-do-i-use-button-when-pressed-and-button-when-held-together

**Reply by Torxsmind** [reply]
I missed that, let me test.  Thanks
